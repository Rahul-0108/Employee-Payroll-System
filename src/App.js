import "./App.css";
import EmployeeUI from "./EmployeeUI";
import { Provider } from "react-redux";
import store from "./Redux/store";

function App() {
  return (
    <Provider store={store}>
      <div>
        <EmployeeUI />
      </div>
    </Provider>
  );
}

export default App;
