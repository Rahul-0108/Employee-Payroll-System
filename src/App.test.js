// This is just a demonstration of Snapshot test and React test Library

import { render, screen } from "@testing-library/react";
import renderer from "react-test-renderer";
import App from "./App";

describe("App.tsx Test", () => { // groups tests
  test("renders EmployeeUI Component", () => {
    jest.mock("./EmployeeUI", () => () => "EmployeeUI");
    const component = renderer.create(<App />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
    jest.clearAllMocks();
  });
  test("Employee Money System is rendered from EmployeeUI Component", () => {
    render(<App />);
    expect(screen.queryByText("Employee Money System")).toBeInTheDocument();
  });
});
