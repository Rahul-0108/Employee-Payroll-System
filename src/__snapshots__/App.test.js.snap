// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App.tsx Test renders EmployeeUI Component 1`] = `
<div>
  <div
    style={
      Object {
        "textAlign": "center",
      }
    }
  >
    <h1
      style={
        Object {
          "backgroundColor": "black",
          "color": "white",
          "padding": "8px",
        }
      }
    >
      Employee Money System
    </h1>
    <table
      className="table-bordered"
      style={
        Object {
          "marginLeft": "auto",
          "marginRight": "auto",
          "marginTop": "8%",
          "width": "50%",
        }
      }
    >
      <thead>
        <tr>
          <th>
            EmployeeId
          </th>
          <th>
            Account Total Money
          </th>
          <th>
            Transaction Amount
          </th>
          <th>
            Add
          </th>
          <th>
            Remove
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td>
            Employee1
          </td>
          <td>
            1000
          </td>
          <td>
            <input
              className="form-control"
              id="id1"
              onChange={[Function]}
              type="text"
              value={0}
            />
          </td>
          <td>
            <button
              onClick={[Function]}
              style={
                Object {
                  "borderWidth": "0px",
                }
              }
            >
              Add
            </button>
          </td>
          <td>
            <button
              onClick={[Function]}
              style={
                Object {
                  "borderWidth": "0px",
                }
              }
            >
              Remove
            </button>
          </td>
        </tr>
        <tr>
          <td>
            Employee2
          </td>
          <td>
            1000
          </td>
          <td>
            <input
              className="form-control"
              id="id2"
              onChange={[Function]}
              type="text"
              value={0}
            />
          </td>
          <td>
            <button
              onClick={[Function]}
              style={
                Object {
                  "borderWidth": "0px",
                }
              }
            >
              Add
            </button>
          </td>
          <td>
            <button
              onClick={[Function]}
              style={
                Object {
                  "borderWidth": "0px",
                }
              }
            >
              Remove
            </button>
          </td>
        </tr>
        <tr>
          <td>
            Employee3
          </td>
          <td>
            1000
          </td>
          <td>
            <input
              className="form-control"
              id="id3"
              onChange={[Function]}
              type="text"
              value={0}
            />
          </td>
          <td>
            <button
              onClick={[Function]}
              style={
                Object {
                  "borderWidth": "0px",
                }
              }
            >
              Add
            </button>
          </td>
          <td>
            <button
              onClick={[Function]}
              style={
                Object {
                  "borderWidth": "0px",
                }
              }
            >
              Remove
            </button>
          </td>
        </tr>
        <tr>
          <td>
            Employee4
          </td>
          <td>
            1000
          </td>
          <td>
            <input
              className="form-control"
              id="id4"
              onChange={[Function]}
              type="text"
              value={0}
            />
          </td>
          <td>
            <button
              onClick={[Function]}
              style={
                Object {
                  "borderWidth": "0px",
                }
              }
            >
              Add
            </button>
          </td>
          <td>
            <button
              onClick={[Function]}
              style={
                Object {
                  "borderWidth": "0px",
                }
              }
            >
              Remove
            </button>
          </td>
        </tr>
        <tr>
          <td>
            Employee5
          </td>
          <td>
            1000
          </td>
          <td>
            <input
              className="form-control"
              id="id5"
              onChange={[Function]}
              type="text"
              value={0}
            />
          </td>
          <td>
            <button
              onClick={[Function]}
              style={
                Object {
                  "borderWidth": "0px",
                }
              }
            >
              Add
            </button>
          </td>
          <td>
            <button
              onClick={[Function]}
              style={
                Object {
                  "borderWidth": "0px",
                }
              }
            >
              Remove
            </button>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
